Tasks/UseDotNetV2/versionfetcher.ts

```diff
@@ -13,7 +13,9 @@ import { VersionInfo, Channel, VersionFilesData, VersionParts } from "./models"
 import * as utils from "./versionutilities";
 
 export class DotNetCoreVersionFetcher {
-    constructor() {
+    private explicitVersioning: boolean = false;
```

Comment:
We should also consider caching the following as they will help in multiple versions to be downloaded using globalJson (we are already caching the releases-index.json content and storing channel information)

1. osDetails being fetched [HERE](https://github.com/microsoft/azure-pipelines-tasks/pull/10734/files?file-filters%5B%5D=.md&file-filters%5B%5D=.ts#diff-5952be6806f8423fdfa8986c5c93b42eL59) should be cached

2. releases.json version information being fetched [HERE](https://github.com/microsoft/azure-pipelines-tasks/pull/10734/files?file-filters%5B%5D=.md&file-filters%5B%5D=.ts#diff-5952be6806f8423fdfa8986c5c93b42eR150), can be cached for the last major version accessed.

Reply:
On further thought this seems to be an additional feature which can be added later. For tracking it separately i have filed and assigned it to myself: #10885 

So, this PR is good to go without catering to caching now.

Reply:
resolved because there is a new feature.